### PR TITLE
Inline Constant Mutator

### PR DESCRIPTION
**Pull request by filipevarjao**
This PR adds the [Inline Constant Mutator](https://pitest.org/quickstart/mutators/#INLINE_CONSTS) where a boolean is replace the unmutated value true with false and replace the unmutated value false with true

### CODE REVIEW

File: test/fixtures/hello_world.ex [hunk 1]
```diff
@@ -9,4 +9,7 @@ defmodule HelloWorld do
       :no
     end
   end
+
+  def lie_to_me?(true), do: true
+  def lie_to_me?(false), do: false
```
Comment (dnlserrano) [inline test/fixtures/hello_world.ex:14]: I think you'd need to test this in `hello_world_test.exs` in order for it to be exercised as part of `mix exavier.test`? Something along the lines of:

```elixir
  test "when lie_to_me? true" do
    assert @subject.lie_to_me?(true) == true
  end

  test "when lie_to_me? false" do
    assert @subject.lie_to_me?(false) == false
  end
```

If you then run `MIX_ENV=test mix exavier.test` locally, it revelas one of the current issues of this library (which I might try and address if I have someone else to bounce ideas off of 😛), which is it mutates the whole module various times for each test, which is not ideal. E.g., it can mutate function A (not related to function B which you just added) and counts that as a live mutant (since test will still pass, given change happened elsewhere, not in function B). How `exavier` works is further explained [here](https://github.com/dnlserrano/exavier#how-does-exavier-work), but I can pitch ideas on how we can potentially improve this.

TL;DR: For now, I think it'd be more accurante to include the testing above. Thoughts?

File: test/exavier/mutators/icm1_test.exs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+defmodule Exavier.Mutators.ICM1Test do
+  use ExUnit.Case, async: true
+
+  @subject Exavier.Mutators.ICM1
+
+  @mutations [
+    %{
+      description: "mutates true to false",
+      original_code: {true, [line: 2], [1, 1]},
```
Comment (dnlserrano) [inline test/exavier/mutators/icm1_test.exs:9]: I don't think this is the exact structure of an original quoted code using something like what you describe in the added mutator? Example code for a module containing only the `lie_to_me?/1` functions, with an assignment and return of that value (given just returning `true` or `false` would cause the `cover`age tool from Erlang to not pick up on test-covered code):

```elixir
defmodule HelloWorld do
  def lie_to_me?(true) do
    lying = true
    lying
  end

  def lie_to_me?(false) do
    lying = false
    lying
  end
end
```

```elixir
{:defmodule, [line: 1],
 [
   {:__aliases__, [line: 1], [:HelloWorld]},
   [
     do: {:__block__, [],
      [
        {:def, [line: 2],
         [
           {:lie_to_me?, [line: 2], [true]},
           [
             do: {:__block__, [],
              [
                {:=, [line: 3], [{:lying, [line: 3], nil}, true]},
                {:lying, [line: 4], nil}
              ]}
           ]
         ]},
        {:def, [line: 7],
         [
           {:lie_to_me?, [line: 7], [false]},
           [
             do: {:__block__, [],
              [
                {:=, [line: 8], [{:lying, [line: 8], nil}, false]},
                {:lying, [line: 9], nil}
              ]}
           ]
         ]}
      ]}
   ]
 ]}
```

What are your thought on the mutator? Would you like to add it for assignments only for now or tweak it to also handle arbitrary occurrences in other places in the code (e.g., `condition1 && condition2 && false`) although maybe this example doesn't bring much value, since constant booleans don't usually pop up like that?

Anyway, if we're considering assignments I'd change this to:

```elixir
description: "mutates true to false",
original_code: {:=, [line: 1], [{:var, [line: 1], nil}, true]},
lines_to_mutate: [1],
mutated_code: {:=, [line: 1], [{:var, [line: 1], nil}, false]},
```

Something along those lines.

File: lib/exavier/mutators/icm1.ex [hunk 3]
```diff
@@ -0,0 +1,24 @@
+defmodule Exavier.Mutators.ICM1 do
+  @behaviour Exavier.Mutators.Mutator
+
+  @mutations %{
+    true => false,
+    false => true
+  }
```
Comment (dnlserrano) [inline lib/exavier/mutators/icm1.ex:7]: Because of they way we'd want to do this (possibly in assignment, IIUC), I'd suggest this alternative implementation:

```elixir
defmodule Exavier.Mutators.ICM1 do
  @behaviour Exavier.Mutators.Mutator

  @operator :=

  @impl Exavier.Mutators.Mutator
  def operators, do: [@operator]

  @impl Exavier.Mutators.Mutator
  def mutate(ast, lines_to_mutate) do
    do_mutate(ast, lines_to_mutate)
  end

  defp do_mutate({nil, _, _}, _), do: :skip

  defp do_mutate({@operator, meta, [var, true]}, _lines_to_mutate) do
    {@operator, meta, [var, false]}
  end

  defp do_mutate({@operator, meta, [var, false]}, _lines_to_mutate) do
    {@operator, meta, [var, true]}
  end

  defp do_mutate(_, _), do: :skip
end
```

Let me know if you think it (doesn't) makes sense.

**Comment by dnlserrano** [comment]
Thanks for your interest in `exavier` @filipevarjao! 🤓 I left some comments on your MR. Looking forward to have this added into our lib, to make even better, with more diverse mutants! 🦎

**Comment by filipevarjao** [comment]
hi @dnlserrano I'm new to the concept of mutation test, very interesting, I see it can reveal the issues on the test suite, this PR was my first attempt, but I'm still trying to figure out how exavier works. Thanks for your comments I'll fix them, and it will help me to understand it better. Then I'll be more able to contribute with new approaches and ideas.

**Comment by dnlserrano** [comment]
Closing this for now. Feel free to re-open with discussed improvements/feedback incorporated anytime. Thanks for your interest once again!
